scripts/migration/migrate-addon.sh

```diff
@@ -0,0 +1,653 @@
+    "image":"gcr.io/gke-release/istio/pilot:1.4.10-gke.12",
+    "env":[{"name":"PILOT_SKIP_VALIDATE_TRUST_DOMAIN","value":"true"}]
+  }]}}}}'
+
+  kube patch deploy istio-citadel -n istio-system  -p='{"spec":{"template":{"spec":{
+    "containers":[{
+      "name":"citadel",
+      "args": ["--append-dns-names=true", "--grpc-port=8060", "--citadel-storage-namespace=istio-system", "--custom-dns-names=istio-pilot-service-account.istio-system:istio-pilot.istio-system", "--monitoring-port=15014", "--self-signed-ca=true", "--workload-cert-ttl=2160h", "--root-cert=/var/run/root-certs/meshca-root.pem"],
+      "volumeMounts": [{"mountPath": "/var/run/root-certs", "name": "meshca-root", "readOnly": true}]
+    }],
+    "volumes": [{"name": "meshca-root", "secret":{"secretName": "meshca-root"}}]
+  }}}}'
+  echo -e "${green}OK${clr}"
+
+  # Wait for Citadel to distribute the cert everywhere. Note: this always takes at least one minute
+  # due to a bug/feature in Citadel causing it to delay provisioning at startup.
+  heading "Waiting for root certificate to distribute to all pods. This will take a few minutes..."
+  for namespace in $(kubectl get namespaces -oname | cut -d/ -f2); do
+    # Check against a portion of the root to see if its been distributed.
+    # Note: this doesn't ensure the workloads have actually hot-restarted to pickup the change.
+    ROOT_CERT="$(kubectl get secret -n "${namespace}" istio.default -ojsonpath='{.data.root-cert\.pem}')"
+    LAST_ERR=$?
+    if [[ "${LAST_ERR}" != 0 ]]; then
+      # Likely namespace was deleted while in progress
+      continue
+    fi
+    while ! echo "${ROOT_CERT}" | base64 -d | grep -q 8zyiCcMm1i2yVVphwE1kczFwunAh0JB896VaXGVxXeKEAMQoXHjgDdCYp8; do
+      echo "ASM root certificate not distributed to ${namespace}, trying again later"
+      sleep 3
+      ROOT_CERT="$(kubectl get secret -n "${namespace}" istio.default -ojsonpath='{.data.root-cert\.pem}')"
+      LAST_ERR=$?
+      if [[ "${LAST_ERR}" != 0 ]]; then
+        # Likely namespace was deleted while in progress
+        continue
+      fi
+    done
+    echo -e "${green}ASM root certificate distributed to namespace ${namespace}${clr}"
+  done
+  # Wait 75 seconds. Kubelet runs a sync loop on a 1 minute interval, so its unlikely to take more than 75s
+  # ideally we have a better wait here, but its hard to do. We can exec in and read the file or hit /certs,
+  # but both would be slow and require a lot of code to write.
+  if [[ "${SKIP_WAIT_FOR_ROOT:-}" == "" ]]; then
+    echo "Waiting for proxies to pick up the new root certificate..."; sleep 15
+    echo "Waiting for proxies to pick up the new root certificate..."; sleep 15
+    echo "Waiting for proxies to pick up the new root certificate..."; sleep 15
+    echo "Waiting for proxies to pick up the new root certificate..."; sleep 15
+    echo "Waiting for proxies to pick up the new root certificate..."; sleep 15
+  fi
+  echo -e "${green}OK${clr}"
+}
+
+disable_galley_webhook() {
```

Comment:
`replace_galley_webhook` seems like a better name

Reply:
This actually is just disabling the galley webook. Will change the header message.